app/org/thp/cortex/services/DockerJobRunnerSrv.scala

```diff
@@ -50,89 +51,115 @@ class DockerJobRunnerSrv(
-    config.getOptional[Seq[String]]("docker.container.dns").map(_.asJava).foreach(hostConfigBuilder.dns)
-    config.getOptional[Seq[String]]("docker.container.dnsSearch").map(_.asJava).foreach(hostConfigBuilder.dnsSearch)
-    config.getOptional[Seq[String]]("docker.container.extraHosts").map(_.asJava).foreach(hostConfigBuilder.extraHosts)
-    config.getOptional[Long]("docker.container.kernelMemory").foreach(hostConfigBuilder.kernelMemory(_))
-    config.getOptional[Long]("docker.container.memoryReservation").foreach(hostConfigBuilder.memoryReservation(_))
-    config.getOptional[Long]("docker.container.memory").foreach(hostConfigBuilder.memory(_))
-    config.getOptional[Long]("docker.container.memorySwap").foreach(hostConfigBuilder.memorySwap(_))
-    config.getOptional[Int]("docker.container.memorySwappiness").foreach(hostConfigBuilder.memorySwappiness(_))
-    config.getOptional[String]("docker.container.networkMode").foreach(hostConfigBuilder.networkMode)
-    config.getOptional[Boolean]("docker.container.privileged").foreach(hostConfigBuilder.privileged(_))
-    hostConfigBuilder.appendBinds(
-      Bind
-        .from(dockerJobBaseDirectory.resolve(jobBaseDirectory.relativize(jobDirectory)).toAbsolutePath.toString)
-        .to("/job")
-        .readOnly(false)
-        .build()
-    )
-    val cacertsFile = jobDirectory.resolve("input").resolve("cacerts")
-    val containerConfigBuilder = ContainerConfig
-      .builder()
-      .hostConfig(hostConfigBuilder.build())
-      .image(dockerImage)
-      .cmd("/job")
+  private def generateErrorOutput(containerId: String, f: Path) = {
+    logger.warn(s"the runner didn't generate any output file $f")
+    for {
+      output <- javaClient.getLogs(containerId)
+      report = Json.obj("success" -> false, "errorMessage" -> output)
+      _ <- Try(Files.write(f, report.toString.getBytes(StandardCharsets.UTF_8)))
+    } yield report
+  }
 
-    val containerConfig =
-      if (Files.exists(cacertsFile)) containerConfigBuilder.env(s"REQUESTS_CA_BUNDLE=/job/input/cacerts").build()
-      else containerConfigBuilder.build()
-    val containerCreation = client.createContainer(containerConfig)
-    //          Option(containerCreation.warnings()).flatMap(_.asScala).foreach(logger.warn)
+  def run(jobDirectory: Path, dockerImage: String, timeout: Option[FiniteDuration])(implicit executionContext: ExecutionContext): Try[Unit] = {
+    val to = timeout.getOrElse(FiniteDuration(5000, TimeUnit.SECONDS))
 
-    logger.debug(s"Container configuration: $containerConfig")
-    logger.info(
-      s"Execute container ${containerCreation.id()}\n" +
-        s"  timeout: ${timeout.fold("none")(_.toString)}\n" +
-        s"  image  : $dockerImage\n" +
-        s"  volume : ${jobDirectory.toAbsolutePath}:/job" +
-        Option(containerConfig.env()).fold("")(_.asScala.map("\n  env    : " + _).mkString)
-    )
+    if (autoUpdate) Try(javaClient.pullImage(dockerImage, to))
```

Comment:
The analyzer timeout is not related to image download. It should not be used here.
The timeout used to pull image may be configurable (in application.conf)